### ISSUE TITLE
Exclusion by group in segment is not working correctly for all the experiments

### ISSUE THREAD

**Issue by VivekFitkariwala** [CarnegieLearningWeb/UpGrade#1318]
**Describe the bug**

When the user gets excluded due to groups in the segments the exclusion documents are not getting created properly.

Currently, we are creating a group exclusion document for all the experiment types that don't make sense for Individual and within-subject experiments. For group experiments, this document should only be created if the `experiment.group` is the same as the group type because of which the user is excluded. Else only an individual exclusion document should be created for the user.

**To summarise**
Individual/Within-subject experiment
- Only individual exclusion documents should be created.

Group Experiment
- Individual Exclusion documents should be created.
- A group Exclusion document should be created only if the experiment group is the same as the segment's group because of which the user got excluded

**Comment by bcb37** [comment]
QA: @ppratikcr7  See my comment on [the other issue](https://github.com/CarnegieLearningWeb/UpGrade/issues/1301#issuecomment-2000400411)
